src/ocean/task/internal/FiberPoolWithQueue.d

```diff
@@ -0,0 +1,241 @@
+/*******************************************************************************
+
+    Extends reusable fiber pool with task queue. Used in task scheduler
+    as "default" way to schedule tasks.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.task.internal.FiberPoolWithQueue;
+
+import core.thread : Fiber;
+
+import ocean.meta.types.Qualifiers;
+import ocean.core.Enforce;
+import ocean.task.internal.FiberPool;
+import ocean.task.Task;
+import ocean.util.container.queue.FixedRingQueue;
+
+debug (TaskScheduler)
+    import ocean.io.Stdout;
+
+/// Ditto
+public class FiberPoolWithQueue : FiberPool
+{
+    /// Queue of tasks awaiting idle worker fiber to be executed in
+    /* package(ocean.task) */
+    public FixedRingQueue!(Task) queued_tasks;
+
+    /// Thrown when all fibers are busy, task queue is full and no custom
+    /// delegate to handle queue overflow is supplied.
+    /* package(ocean.task) */
+    public TaskQueueFullException queue_full_e;
+
+    /// Reference to `theScheduler.processEvents` initialized upon class
+    /// creation. Used to break circular dependency.
+    private void delegate(istring, int) process_events;
+
+    /***************************************************************************
+
+        Called each time task is attempted to be queue but size limit is
+        reached. Both the queue and the task will be supplied as arguments.
+
+    ***************************************************************************/
+
+    public TaskQueueFullCB task_queue_full_cb;
+
+    /**************************************************************************
+
+        Constructor
+
+        Params:
+            stack_size = fiber stack size to use in this poll
```

Comment:
Incomplete params doc.

Reply:
Still incomplete.

Reply:
https://github.com/sociomantic-tsunami/ocean/pull/293/files#diff-2c0e70ae37423c24132186053ded1353R68

Reply:
Ah, the comment fix was in the wrong commit.